code/script.js

```diff
@@ -36,12 +44,183 @@ const greeting = () => {
+   
+}
+
+//Question 2
+
+const moodOptions = () => {
+  let userName = document.getElementById('user-input');
+  showMessage(`How are you feeling today?`, 'bot')
+  inputWrapper.innerHTML = `
+    <button id="option1">happy</button>
+    <button id="option2">sad</button>
+    <button id="option3">angry</button>
+  `
+  document.getElementById("option1").addEventListener("click", () => {
+    showMessage("I'm happy", "user")
+  })
+  document.getElementById("option2").addEventListener("click", () => {
+    showMessage("I'm sad", "user")
+  })
+  document.getElementById("option3").addEventListener("click", () => {
+    showMessage("I'm angry", "user")
+  })
+  
+  document.getElementById("option1").addEventListener("click", () => {
+    showMessage("I see you are feeling happy!😊 Here's a list that suits your mood. Go ahead and choose one! 👇", "bot")
+    inpt = () => { 
+      inputWrapper.innerHTML= `
+        <select id="select" name="songs">
+          <option value selected disabled>Select a song</option>
+          <option id="select">Happy by William Farrel</option>
+          <option id="select">Here comes the sun by the Beatles</option>
+          <option id="select">Shiny happy people by REM</option>
+        </select>
+      `
+      const select = document.getElementById("select");
+      let userName = document.getElementById('user-input');
+      select.addEventListener("change", () =>
+      greatChoice(select.value))
+    }
+  })
+
+  document.getElementById("option2").addEventListener("click", () => {
+    showMessage("I see that you are sad today😢. Here's a list that suits your mood. Go ahead and choose one! 👇" , "bot")
+    inpt = () => {
+      inputWrapper.innerHTML= `
+        <select id="select" name="songs">
+          <option value selected disabled>Select a song</option>
+          <option id="select">Don't worry, be happy by Bobby Mac Ferrin</option>
+          <option id="select">Someone like you by Adele</option>
+          <option id="select">Hey Jude by The Beatles</option>
+        </select>
+      ` 
+      const select = document.getElementById("select");
+      select.addEventListener("change", () =>
+      greatChoice(select.value))
+    }
+  })
+
+  document.getElementById("option3").addEventListener("click", () => {
+      showMessage("I see that you are angry today😡. Here's a list that suits your mood. Go ahead and choose one! 👇", "bot")
+      inpt = () => {
+        inputWrapper.innerHTML=`
+          <select id="select" name="songs">
+            <option value selected disabled>Select a song</option>
+            <option id="">In the end by Linkin Park</option>
+            <option id="">Basket case by Green Day</option>
+            <option id="">Break Stuff by Limp Bizkit</option>
+         </select>
+        `
+        const select = document.getElementById("select");
+        select.addEventListener("change", () =>
+        greatChoice(select.value))
+      }   
+   })
```

Comment:
all 3 events have the same skeleton ☠️ This is a great opportunity to extract this logic to the function and call it when needed. 😸

Reply:
Can you leave an example on how to do that? I'm not entirely sure of what you mean.